enumerables.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each
+    length.times do |v|
+      yield(self[v])
+    end
+  end
+
+  def my_each_with_index
+    length.times do |v|
+      yield(self[v], v)
+    end
+  end
+
+  def my_select
+    arr = []
+    length.times do |v|
+      arr.push(self[v]) if yield(self[v])
+    end
+    arr
+  end
+
+  def my_all?
+    result = true
+    length.times do |v|
+      result = false if yield(self[v]) != true
+      break unless result == true
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    length.times do |v|
+      result = true if yield(self[v])
+      break unless result == false
+    end
+    result
+  end
+
+  def my_none?
+    result = true
+    length.times do |v|
+      result = false if yield(self[v])
+      break unless result == true
+    end
+    result
+  end
+
+  def my_count
+    result = 0
+    length.times do |v|
+      result += 1 if yield(self[v])
+    end
+    result
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns the number of items in enum through enumeration
```ruby
array = Array.new(100){rand(0...9)}
array.my_count == array.count #true
```
- [ ] counts the number of items in enum that are equal to item if an argument is given
```ruby
array.my_count(0) == array.count(0) #true
```
- [ ] counts the number of elements yielding a true value if a block is given
```ruby
array.my_count(&block) == array.count(&block)
```